presto-main/src/main/java/io/prestosql/execution/scheduler/TopologyAwareNodeSelectorModule.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.execution.scheduler;
+
+import com.google.inject.Binder;
+import com.google.inject.Module;
+import com.google.inject.Scopes;
+
+import static io.airlift.configuration.ConfigBinder.configBinder;
+
+public class TopologyAwareNodeSelectorModule
+        implements Module
+{
+    @Override
+    public void configure(Binder binder)
+    {
+        configBinder(binder).bindConfig(NetworkTopologyConfig.class);
+        binder.bind(NetworkTopology.class).to(FlatNetworkTopology.class).in(Scopes.SINGLETON);
```

Comment:
@oneonestar I might be misunderstanding your concern, but config is actually why you want to use modules. Each module can bind a different config object. This means that the config is conditional based on which topology is used, rather than having to make them all available at once. We can conditionally install modules due to `AbstractConfigurationAwareModule` which works with `Bootstrap` to make configuration available at module configuration time. `HiveSecurityModule` is a good example of this.

Reply:
@electrum Thanks for the pointer. I tried to use the conditional binding. Is this correct?
https://github.com/prestosql/presto/pull/987/commits/7421b3c877ab94be3b9d4e38b969c382db0425b5#diff-4cf17cc6e9dd4b9298395790e0d2742c